tests/unit/test_synapse_api.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Synapse API unit tests."""
+
+# pylint: disable=protected-access
+
+import typing
+from secrets import token_hex
+from unittest import mock
+
+import pytest
+import requests
+
+import synapse
+from user import User
+
+
+def test_register_user_success(monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: set User parameters.
+    act: register the user.
+    assert: parameters are passed correctly.
+    """
+    username = "any-user"
+    user_data: dict[str, typing.Any] = {
+        "username": username,
+        "admin": "yes",
+    }
+    user = User(**user_data)
+    get_nonce_return = "nonce"
+    get_nonce_mock = mock.MagicMock(return_value=get_nonce_return)
+    monkeypatch.setattr("synapse.api._get_nonce", get_nonce_mock)
+    generate_mac_mock = mock.MagicMock(return_value="mac")
+    monkeypatch.setattr("synapse.api._generate_mac", generate_mac_mock)
+    mock_response = mock.Mock()
+    mock_response.raise_for_status.return_value = None
+    monkeypatch.setattr("synapse.api.requests.post", mock_response)
+    shared_secret = token_hex(16)
+    synapse.register_user(shared_secret, user)
```

Comment:
Would it be worth separating the arrange, act and assert blocks using a new line? To make it clearer when the setup, execution and checks are being done?

Reply:
Sure, I added comments now.